bigquery/snippets/job/bigquery_copy_table.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package job
+
+// [START bigquery_copy_table]
+
+import (
+	"cloud.google.com/go/bigquery"
+	"context"
+	"fmt"
+)
+
+func copyTable(projectID, datasetID, srcID, dstID string) error {
+	// projectID := "my-project-id"
+	// datasetID := "mydataset"
+	// srcID := "sourcetable"
+	// dstID := "destinationtable"
+	ctx := context.Background()
+	client, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("bigquery.NewClient: %v", err)
+	}
+	dataset := client.Dataset(datasetID)
+	copier := dataset.Table(dstID).CopierFrom(dataset.Table(srcID))
+	copier.WriteDisposition = bigquery.WriteTruncate
+	job, err := copier.Run(ctx)
+	if err != nil {
+		return err
```

Comment:
Add error context with `fmt.Errorf`? Looks like in other places, too.

Reply:
I'm somewhat hesitant to do a bunch of extra error annotation/propagation in the general samples.  I think such a thing is more suitable for our troubleshooting topics, which does not (yet) have samples.  By not doing this it also makes it easier for users to get at the [structured errors](https://godoc.org/cloud.google.com/go/bigquery#hdr-Errors)